scripts/private/_sanity-check-universal.js

```diff
@@ -64,53 +75,63 @@ async function main() {
       console.log(`❌ Error getting dispatcher address from Universal Channel Mw or from config: ${error}`);
       return;
     }
-  } else {
+  }
+
+  // If sanity check is false, log an error and return
+  if (!sanityCheck) {
     console.log(`
 ⛔ Sanity check failed for network ${networkName}, 
 check if the values provided in the .env file for the Universal Channel Mw and the dispatcher are correct.
 --------------------------------------------------
-🔮 Expected Universal Channel Handler (in IBC contract): ${ucHandlerAddr}...
-🗃️  Found Universal Channel Handler (in .env file): ${envUcHandlerAddr}...
+🔮 Expected Dispatcher (in Universal Channel Handler contract): ${dispatcherAddr}...
+🗃️  Found Dispatcher (in .env file): ${envDispatcherAddr}...
 --------------------------------------------------
-        `);
+    `);
     return;
-  }
+  } else {
+    // If the sanity check passes, we can continue to check the channel ID stored in the Universal Channel Mw
+    let counter = 0;
+    let channelId, envChannelId;
+    let channelBytes;
+    let foundChannel = true;
+    // We don't know how many channels are connected to the Universal Channel Mw, so we loop until we get an error
+    do {
+      // Try to get the channel ID at the index
+      try {
+        channelBytes = await ucHandler.connectedChannels(counter);
+      } catch (error) {
+        // If we get an error, it means we reached the end of the list, do not return, just log the error and set foundChannel to false
+        // console.log(`❌ No channel ID at index: ${counter}`);
+        foundChannel = false;
+      }
+      counter++;
+    } while (foundChannel);
 
-  if (sanityCheck) {
-    const channelBytes = await ucHandler.connectedChannels(0);
-    const channelId = hre.ethers.decodeBytes32String(channelBytes);
-    const envChannelId = config['sendUniversalPacket'][networkName]['channelId'];
+    // channelBytes should be the last (populated) index in the connectedChannels array
+    channelId = hre.ethers.decodeBytes32String(channelBytes);
+    console.log(`Channel ID in UCH contract: ${channelId}`);
+    envChannelId = config['sendUniversalPacket'][networkName]['channelId'];
 
-    if (channelId !== envChannelId) {
+    // Compare the channel ID with the one in the .env file and log an error if they don't match
+    // Run only after we've encountered an error fetching a channel ID at a new index
+    if (channelId === undefined && channelId !== envChannelId) {
       sanityCheck = false;
       console.log(`
 ⛔ Sanity check failed for network ${networkName}, 
 check if the channel id value for the Universal channel in the config is correct.
 --------------------------------------------------
 🔮 Expected Channel ID (in Universal Channel Handler contract): ${channelId}...
-🗃️  Found Dispatcher (in .env file): ${envChannelId}...
+🗃️  Found Channel ID (in config file): ${envChannelId}...
 --------------------------------------------------
-`);
+  `);
```

Comment:
I agree this would be better, but for now the contract is to be taken as is and it does not include any other useful functions like `channelExists`. The script itself is a helper script however, so if there were a different error than the one we expect that is caught in the try catch, it still does not completely break the flow.